### ISSUE TITLE
Ejabberd REST API (privacy_set) is not working

### ISSUE THREAD

**Issue by aungthawaye** [processone/ejabberd#2205]
> What version of ejabberd are you using?
ejabber 17.12

> What operating system (version) are you using?

Mac OS X 10.13.2 (High Sierra)

> How did you install ejabberd (source, package, distribution)?

Home brew. (I cannot use Mac OS Ejabberd installer. It shows error when I startup server.  _Incompatible library version: fxml_stream.so requires version 8.0.0 or later, but libexpat.1.dylib provides version 7.0.0'_)

> What did not work as expected? Are there error messages in the log? What
> was the unexpected behavior? What was the expected result?

Previously, I was running ejabberd 17.01 on Mac OS Sierra. Then I upgraded ejabberd to 17.12 and also macOS to High Sierra. After I have upgraded, I noticed I cannot call privacy_set REST API. It returns success "0". But no data was inserted into MySQL db. MySQL version is 5.7.

**But REST APIs (add_roster & register) are working fine.** 

Below is my REST API (privacy_set)'s request JSON :

```
{
  "user": "myusername",
  "host": "myhost",
  "xmlquery": "<query xmlns='jabber:iq:privacy'><list name='FRIENDS'><item type='group' value='FRIENDS' action='allow' order='4'><message/></item></list></query>"
}
```
HTTP Request :

```
POST /api/privacy_set HTTP/1.1
Host: 127.0.0.1:5280
Content-Type: application/json
Authorization: Basic **<<MY_ADMIN_AUTH_VALUE>>**
Cache-Control: no-cache
Postman-Token: d3737449-8972-4eb6-5e65-75b33d296a68

{
  "user": "myusername",
  "host": "myhost",
  "xmlquery": "<query xmlns='jabber:iq:privacy'><list name='FRIENDS'><item type='group' value='FRIENDS' action='deny' order='4'><message/></item></list></query>"
}
```

Above request JSON was working fine in ejabberd 17.01. Now in 17.12, it doesn't work. 

Please kindly help because I need to launch my application sooner, and suddenly face this issue.

**Comment by prefiks** [comment]
What happens when you do that? Are you getting error response? Could you check server logs for errors?

**Comment by aungthawaye** [comment]
Hi, 

I get SUCCESS response which is "0" from REST API (as per API documentation). And I was monitoring using **ejabberdctl live** and here is what I saw whenever I run it. I am sure there was no error.

```
15:41:30.720 [debug] (#Port<0.16958>) http query: 'POST' <<"/api/privacy_set">>
15:41:30.720 [debug] client data: <<"{\n  \"user\": \"aungthawaye\",\n  \"host\": \"im.elissa.software\",\n  \"xmlquery\": \"<query xmlns='jabber:iq:privacy'><list name='FRIENDS'><item type='group' value='FRIENDS' action='allow' order='4'><message/></item></list></query>\"\n}\n\n">>
15:41:30.720 [debug] [<<"api">>,<<"privacy_set">>] matches [<<"api">>]
15:41:30.720 [info] API call privacy_set [{<<"user">>,<<"aungthawaye">>},{<<"host">>,<<"im.elissa.software">>},{<<"xmlquery">>,<<"<query xmlns='jabber:iq:privacy'><list name='FRIENDS'><item type='group' value='FRIENDS' action='allow' order='4'><message/></item></list></query>">>}] from ::ffff:127.0.0.1:50750
15:41:30.720 [debug] Command 'privacy_set' execution allowed by rule 'admin access' (CallerInfo=#{caller_module => mod_http_api,caller_server => <<"im.elissa.software">>,ip => {0,0,0,0,0,65535,32512,1},usr => {<<"admin">>,<<"im.elissa.software">>,<<>>}})
15:41:30.720 [debug] Executing command mod_admin_extra:privacy_set with Args=[<<"aungthawaye">>,<<"im.elissa.software">>,<<"<query xmlns='jabber:iq:privacy'><list name='FRIENDS'><item type='group' value='FRIENDS' action='allow' order='4'><message/></item></list></query>">>]
```

**Comment by prefiks** [comment]
Should be fixed by commit 4c799528c79296ec2a2ad5283854280c5c64a5e1
